File: Modul 6. Array/PRAK604/PRAK604-2310817120003-NatalieGraceKatiandagho.py
kode = input()
pesan = input()

panjang_kode = len(kode)
panjang_pesan = len(pesan)

if panjang_kode != panjang_pesan:
    print("Panjang kalimat berbeda, pesan palsu")
    exit()

bintang = 0
pagar = 0

hasil_pesan = ""

for i in range(panjang_kode):
    if kode[i] == ' ' and pesan[i] == ' ':
        hasil_pesan += ' '
    elif kode[i] == pesan[i]:
        hasil_pesan += '*'
        bintang += 1
    else:
        hasil_pesan += '#'
        pagar += 1

print(hasil_pesan)
print(f"* = {bintang}")
print(f"# = {pagar}")

if bintang >= pagar:
    print("Pesan asli")
else:
    print("Pesan palsu")

File: Modul 6. Array/PRAK604/PRAK604-2310817120003-NatalieGraceKatiandagho.c
#include <stdio.h>
#include <string.h>

int main()
{
    char kode[100];
    char pesan[100];
    int i;
    int bintang = 0;
    int pagar = 0;

    scanf(" %[^\n]", kode);
    scanf(" %[^\n]", pesan);

    int panjangKode = strlen(kode);
    int panjangPesan = strlen(pesan);

    if (panjangKode != panjangPesan)
    {
        printf("Panjang kalimat berbeda, pesan palsu\n");
        return 0;
    }

    for (int i = 0; i < panjangKode; i++)
    {
        if (kode[i] == ' ' && pesan[i] == ' ')
        {
            pesan[i] = ' ';
        }
        else if (kode[i] == pesan[i])
        {
            pesan[i] = '*';
        }
        else
        {
            pesan[i] = '#';
        }
    }

    for (int i = 0; i < panjangKode; i++)
    {
        if (pesan[i] == '*')
        {
            bintang++;
        }
        else if (pesan[i] == '#')
        {
            pagar++;
        }
    }

    printf("%s\n", pesan);
    printf("* = %d\n", bintang);
    printf("# = %d\n", pagar);

    if (bintang >= pagar)
    {
        printf("Pesan asli\n");
    }
    else
    {
        printf("Pesan palsu\n");
    }

    return 0;
}